### PR TITLE
Using Chef instead of Shell provisioner

### PR DESCRIPTION
**Pull request by jjasghar**
This uses chef-solo instead of the shell provisioner for the
Vagrantfile.

So it seems we need to figure out a way to pull down the markdowns from somewhere so I can converge this with TK. Maybe pulling it down via git? Then doing the concatenation.
- [ ] Fix the node-pdf printing, it doesn't seem to work
- [x] rubocop it
- [x] Chefspec it
- [x] feeling sassy `.kitchen.yml` it

### CODE REVIEW

File: cookbooks/bootstrap/recipes/default.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#                          - Han Solo
+#
+
+cookbook_file "/tmp/reveal.json" do
+  owner "vagrant"
+  mode "0644"
+  source 'reveal.json'
+end
+
+execute 'Get the markdowns to the correct location' do
```
Comment (jjasghar) [inline cookbooks/bootstrap/recipes/default.rb]: Adding `git` resource to this that pulls down the markdowns then runs the command _maybe_ the correct way to get this done. I'd like to bounce this idea off a couple people first.

File: cookbooks/bootstrap/spec/unit/recipes/default_spec.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
+#
+# Cookbook Name:: bootstrap
+# Spec:: default
+#
+# Copyright (c) 2015 The Authors, All Rights Reserved.
```
Comment (mattray) [inline cookbooks/bootstrap/spec/unit/recipes/default_spec.rb]: Apache por favor

File: README.md [hunk 3]
```diff
@@ -12,8 +21,13 @@ The format is [reveal-md](https://github.com/webpro/reveal-md). You may use Vagr
 * Open up <http://127.0.0.1:1948/index.md> and you're in the presentation.
 * You may download the slides from the last slide or directly from <http://127.0.0.1:1948/cookbook-guide.pdf>
 
-## Building Locally #
-If you prefer to work with reveal-md directly instead of the Vagrant workflow, you can read the shell provisioner section of the [Vagrantfile](Vagrantfile#L8). The steps are
+## Building Locally
+
+If you prefer to work with reveal-md directly instead of the Vagrant workflow,
+you can use `chef-solo` to bootstrap the cookbook. TODO: chef-solo command here.
```
Comment (mattray) [inline README.md]: Please don't merge without fixing the TODO

File: README.md [hunk 4]
```diff
@@ -12,8 +21,13 @@ The format is [reveal-md](https://github.com/webpro/reveal-md). You may use Vagr
...
+
+If you would like to do it by hand, these are the steps:
```
Comment (mattray) [inline README.md]: If you would prefer the manual steps:

File: cookbooks/bootstrap/.rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+AllCops:
```
Comment (mattray) [inline cookbooks/bootstrap/.rubocop.yml]: Was this cargo-culted from somewhere? Seems kinda like overkill. Perhaps the ChefStyle one?

Comment (jjasghar) [inline cookbooks/bootstrap/.rubocop.yml]: Yeah stole it from the `chef-cookbooks` I'll take a look at the ChefStyle.

**Pull request merged**